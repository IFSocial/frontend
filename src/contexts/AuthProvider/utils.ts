import api from '../../services/api';
import { IUser } from './types';

// Salva o usuario no BD local
export function setUserLocalStorage(user: IUser | null) {
  localStorage.setItem('u', JSON.stringify(user));
}

// Checa se existe o usuario no BD local
export function getUserLocalStorage() {
  const json = localStorage.getItem('u');
  if (!json) {
    return null;
  }

  const user = JSON.parse(json);

  return user ?? null;
}

// Faz request do login passando email e senha como payload
export async function LoginRequest(email: string, password: string) {
  try {
    const request = await api.post('/users/signin', { email, password });

    return request.data;
  } catch {
    return null;
  }
}

// Faz request do cadastrp passando nome, email e senha como payload
export async function SignupRequest(
  name: string,
  registration: string,
  sexo: string,
  email: string,
  password: string,
) {
  try {
    const request = await api.post('/users/signup', {
      name,
      registration,
      sexo,
      email,
      password,
    });

    return request.data;
  } catch {
    return null;
  }
}
