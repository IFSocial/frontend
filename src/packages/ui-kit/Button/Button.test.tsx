import React from 'react';

import { Button } from '.';
import { act, fireEvent, render } from '../../test/utils';

describe('Button', () => {
  it('should render correctly', () => {
    const { getByRole } = render(<Button>Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent(/^Button$/);
    expect(buttonElement).toHaveAttribute('type', 'button');
    expect(buttonElement).toBeEnabled();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a primary button', () => {
    const { getByRole } = render(<Button variant="primary">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a secondary button', () => {
    const { getByRole } = render(<Button variant="secondary">Button</Button>);

    const buttonElement = getByRole('button');

    expect(getByRole('button')).toBeInTheDocument();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a small button', () => {
    const { getByRole } = render(<Button size="small">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a medium button', () => {
    const { getByRole } = render(<Button size="medium">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a large button', () => {
    const { getByRole } = render(<Button size="large">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a submit button', () => {
    const { getByRole } = render(<Button type="submit">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveAttribute('type', 'submit');

    expect(buttonElement).toMatchSnapshot();
  });

  it('can have onClick', () => {
    const handleClick = jest.fn();

    const { getByRole } = render(<Button onClick={handleClick}>Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('can have focus', () => {
    const { getByRole } = render(<Button>Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();

    act(() => {
      fireEvent.keyDown(document.body, { key: 'TAB' });
      buttonElement.focus();
    });

    expect(buttonElement).toHaveFocus();

    expect(buttonElement).toMatchSnapshot();
  });
});
