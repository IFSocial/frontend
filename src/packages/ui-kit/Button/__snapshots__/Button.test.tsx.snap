// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can have focus 1`] = `
<button
  class="sc-bcXHqe gGloyY"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a large button 1`] = `
<button
  class="sc-bcXHqe bIukhK"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a medium button 1`] = `
<button
  class="sc-bcXHqe gGloyY"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a primary button 1`] = `
<button
  class="sc-bcXHqe gGloyY"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a secondary button 1`] = `
<button
  class="sc-bcXHqe hbTobl"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a small button 1`] = `
<button
  class="sc-bcXHqe BztYZ"
  type="button"
>
  Button
</button>
`;

exports[`Button should render a submit button 1`] = `
<button
  class="sc-bcXHqe gGloyY"
  type="submit"
>
  Button
</button>
`;

exports[`Button should render correctly 1`] = `
<button
  class="sc-bcXHqe gGloyY"
  type="button"
>
  Button
</button>
`;
